### ISSUE TITLE
io.fabric8.knative.v1.URL Deserialize failed

### ISSUE THREAD

**Issue by hufei** [fabric8io/kubernetes-client#1727]
client version: 4.4.0
address.json:
{
  "url": "http://nlp-addr.baidu.com"
}
```
package com.hufei.knative.learnk8s;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.fabric8.knative.duck.v1beta1.Addressable;
import org.apache.commons.io.IOUtils;
import org.junit.Test;

import java.io.IOException;

public class K8sClient {

    @Test
    public void testAddr() throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        String jsonString = IOUtils.toString(K8sClient.class.getResourceAsStream("/address.json"), "UTF-8");
        Addressable a = mapper.readValue(jsonString, Addressable.class);
        System.out.println(a.getUrl().toString());
    }
}
```

get error:
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `io.fabric8.knative.v1.URL` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('http://nlp-addr.baidu.com')
 at [Source: (String)"{
  "url": "http://nlp-addr.baidu.com"
}"; line: 2, column: 10] (through reference chain: io.fabric8.knative.duck.v1beta1.Addressable["url"])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1343)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1032)
	at com.fasterxml.jackson.databind.deser.ValueInstantiator._createFromStringFallbacks(ValueInstantiator.java:371)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromString(StdValueInstantiator.java:323)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromString(BeanDeserializerBase.java:1373)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:171)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3004)
	at com.hufei.knative.learnk8s.K8sClient.testAddr(K8sClient.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)```

**Comment by rohanKanojia** [comment]
@hufei : Hi, We recently upgraded our Knative and Tekton versions. Could you please check if you're able to reproduce this on 4.7.0?

**Comment by FWiesner** [comment]
I'm able to reproduce it this way: 

1. setup a Kubernetes cluster with Knative Serving and Eventing
2. create a new namespace
3. create a default broker by labeling the namespace with `knative-eventing-injection=enabled`
4. run the following code:

```java
package demo;

import io.fabric8.knative.client.DefaultKnativeClient;
import io.fabric8.kubernetes.client.Config;
import io.fabric8.kubernetes.client.DefaultKubernetesClient;
import io.fabric8.kubernetes.client.KubernetesClient;
import lombok.SneakyThrows;

import java.io.IOException;
import java.net.URISyntaxException;
import java.nio.file.Files;
import java.nio.file.Paths;


public class Main {
    public static void main(String[] args) throws IOException {
        final var client = getKubernetesClient();

        final var knativeClient = new DefaultKnativeClient(client.getConfiguration());
        knativeClient.inNamespace("demo").brokers().list();
    }

    @SneakyThrows({URISyntaxException.class})
    private static KubernetesClient getKubernetesClient() throws IOException {
        var kubecconfigFileContents = Files.readString(Paths.get(Main.class.getClassLoader().getResource("kubeconfig").toURI()));
        var kubeconfig = Config.fromKubeconfig(kubecconfigFileContents);
        return new DefaultKubernetesClient(kubeconfig);
    }
}
```
The error that the original poster reported pops up

**Comment by FWiesner** [comment]
I used Knative client 4.7.1

**Comment by rohanKanojia** [comment]
I think we need to prioritize these knative/tekton related bugs

**Comment by FWiesner** [comment]
Thanks @rohanKanojia

**Comment by FWiesner** [comment]
@rohanKanojia this one is actually a block for the implementation of custom operators that create and update Knative services. Please see:
```
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `io.fabric8.knative.v1.URL` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('http://foo-ksvc.bar-demo.svc.cluster.local')
 at [Source: (BufferedInputStream); line: 1, column: 1194] (through reference chain: io.fabric8.knative.serving.v1alpha1.ServiceList["items"]->java.util.ArrayList[0]->io.fabric8.knative.serving.v1alpha1.Service["status"]->io.fabric8.knative.serving.v1alpha1.ServiceStatus["address"]->io.fabric8.knative.duck.v1alpha1.Addressable["url"])
```

**Comment by rohanKanojia** [comment]
@FWiesner : Thanks a lot for pointing this out. We've prioritized this issue for next sprint:

**Comment by FWiesner** [comment]
@rohanKanojia I haven't checked yet, as I have to come back from vacation yet, but https://github.com/fabric8io/kubernetes-client/pull/2016 should fix this issue for recent Knative versions

**Comment by rohanKanojia** [comment]
Closed by #2030
